apps/status-api/src/controllers/OracleStatusController.ts

```diff
@@ -21,18 +21,31 @@ export class OracleStatusController {
    */
   @Get('/:address')
   async getOracleStatus (@Param('address') oracleAddress: string): Promise<{ status: OracleStatus }> {
-    const oraclePriceFeed: OraclePriceFeed = await this.cachedGet(`oracle-${oracleAddress}`, async () => {
+    return await this.cachedGet(`oracle-${oracleAddress}`, async () => {
       const oracle = await this.client.oracles.getOracleByAddress(oracleAddress)
-      return (await this.client.oracles.getPriceFeed(oracle.id, oracle.priceFeeds[0].token, oracle.priceFeeds[0].currency, 1))[0]
-    }, 5) // cache result for 5 seconds
+      return await this.getAlivePriceFeed(oracle)
+    }, 5) // cache status result for 5 seconds
+  }
+
+  private async getAlivePriceFeed (oracle: Oracle): Promise<{ status: OracleStatus }> {
+    const id = oracle.id
 
-    const nowEpoch = Date.now()
-    const latestPublishedTime = oraclePriceFeed.block.medianTime * 1000
-    const timeDiff = nowEpoch - latestPublishedTime
+    for (const priceFeed of oracle.priceFeeds) {
+      const token = priceFeed.token
+      const currency = priceFeed.currency
+      const oraclePriceFeed = await this.client.oracles.getPriceFeed(id, token, currency, 1)
+      if (oraclePriceFeed.length !== 0) { // check if ticker is not returning any data, else move on to the next ticker
```

Comment:
```suggestion
      if (oraclePriceFeed.length -== 0) { // check if ticker is not returning any data, else move on to the next ticker
      continue
```

inverse logic for reliability?

Reply:
Whenever you want to use !== 0, consider using >=0 instead for those values that couldn't be a negative value to increase code readability